From c31cae562e48e17f0d9dea6e2ee5d24c133e8d1c Mon Sep 17 00:00:00 2001
From: Jaxson Han <jaxson.han@arm.com>
Date: Thu, 18 May 2023 22:07:54 +0800
Subject: [PATCH] arch: arm64: Fix the STACK_INIT logic during the reset

Each core should init their own stack during the reset when SMP enabled,
but do not touch others. The current init results in each core starting
init the stack from the same address which will break others.

Fix the issue by setting a correct start address.

Signed-off-by: Jaxson Han <jaxson.han@arm.com>
Upstream-Status: Submitted
[https://github.com/zephyrproject-rtos/zephyr/pull/58056]
---
 arch/arm64/core/reset.S | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/core/reset.S b/arch/arm64/core/reset.S
index d01c9c8751..4b826701a2 100644
--- a/arch/arm64/core/reset.S
+++ b/arch/arm64/core/reset.S
@@ -155,7 +155,8 @@ primary_core:
 	 * so that we don't clobber any data.
 	 */
 #ifdef CONFIG_INIT_STACKS
-	ldr	x0, =(z_interrupt_stacks)
+	mov_imm	x0, CONFIG_ISR_STACK_SIZE
+	sub	x0, sp, x0
 	sub     x9, sp, #8
 	mov     x10, 0xaaaaaaaaaaaaaaaa
 stack_init_loop:
-- 
2.34.1

